### PR TITLE
Fixing tracking reconstruction

### PR DESCRIPTION
**Pull request by ebrondol**
In order to have tracking reconstruction working again with the new DetId scheme, the first solution is using the concept of detBrothers in the TkDetLayers builders for Phase2OT.
This is supposed to be not the definitive solution - the definitive one would include the possibility of not to use the DetBrothers and instead using the StackGeomDet directly in the MeasurementTracker class.
The rest of the changes in this PR are concerning the DPG tools used in order to study the efficiency of the tracking local reconstruction. It has been introduced the possibility to check if the selection done on DetId bits and the one done using the TkTopology are the same. 
@boudoul @delaere @venturia

### CODE REVIEW

File: DPGAnalysis/SiStripTools/test/OccupancyPlotsTest_phase2_cfg.py [hunk 1]
```diff
@@ -245,20 +259,20 @@
...
-#process.load("Configuration.Geometry.GeometryExtendedPhaseIPixel_cff")
 process.load("Configuration.StandardSequences.Reconstruction_cff")
 
 from SLHCUpgradeSimulations.Configuration.phase2TkCustomsBE5DPixel10D import *
 
 process.load("Configuration.StandardSequences.FrontierConditions_GlobalTag_cff")
+if not options.globalTag :
+  #from Configuration.AlCa.GlobalTag import GlobalTag
+  #process.GlobalTag = GlobalTag(process.GlobalTag, 'auto:upgradePLS3', '')
+  options.globalTag = "DES23_62_V1::All"
```
Comment (venturia) [inline DPGAnalysis/SiStripTools/test/OccupancyPlotsTest_phase2_cfg.py]: You can set the default values of the "options" parameters, when the options parameters are defined at the beginning of the cfg file

**Comment by VinInn** [comment]
@cmsbuild , please test

**Comment by ebrondol** [comment]
The Unit Tests fails because it cannot open a file.
For the Matrix Tests Outputs, I agreed with @boudoul to wait that all the workflows keep into account the new xmls.

**Comment by ebrondol** [comment]
This PR contains commits already in the release. A new PR has been open with just the commits related to the tracking (PR#12918).

**Pull request closed without merging**